### PR TITLE
New FeedbackReason options

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
### Description
This PR adds `FeedbackEvent.FeedbackReason.INCORRECT_PHONE_NUMBER`, and `FeedbackEvent.FeedbackReason.INCORRECT_RESULT_RANK`

Closes https://github.com/mapbox/mapbox-search-android-internal/issues/803

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [x] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

**Comment by kalbaxa** [comment]
@DzmitryFomchyn please correct api check test first

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/analytics/FeedbackEvent.kt [hunk 1]
```diff
@@ -137,6 +137,16 @@ public class FeedbackEvent @JvmOverloads public constructor(
              */
             public const val INCORRECT_LOCATION: String = "incorrect_location"
 
+            /**
+             * Feedback reason for incorrect POI phone number.
+             */
+            public const val INCORRECT_PHONE_NUMBER: String = "incorrect_phone_number"
```
Comment (friendoye) [inline MapboxSearch/sdk/src/main/java/com/mapbox/search/analytics/FeedbackEvent.kt:143]: [nit]

Please note, that new values are not added to the `@StringDef` declaration:

```kotlin
@Retention(AnnotationRetention.SOURCE)
@StringDef(
      INCORRECT_NAME,
      INCORRECT_ADDRESS,
      INCORRECT_LOCATION,
      INCORRECT_PHONE_NUMBER,
      INCORRECT_RESULT_RANK,
      OTHER,
)
```

**Comment by DzmitryFomchyn** [comment]
> @DzmitryFomchyn please correct api check test first

Ah, sorry, didn't run checks before publishing PR. Fixed, thanks!

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/analytics/FeedbackEvent.kt (hunk 1 above)
Comment (DzmitryFomchyn) [reply MapboxSearch/sdk/src/main/java/com/mapbox/search/analytics/FeedbackEvent.kt:147]: Thanks, fixed :)

**Review by kalbaxa — APPROVED** [review]
lgtm

**Pull request merged**